src/main/kotlin/com/exactpro/th2/read/file/common/AbstractFileReader.kt

```diff
@@ -458,10 +472,14 @@ abstract class AbstractFileReader<T : AutoCloseable>(
                     content = contentParser.parse(streamId, source)
                     if (content.isNotEmpty()) {
                         LOGGER.trace { "Read ${content.size} message(s) for $streamId from ${holder.path}" }
+                        holder.readState = FileHolder.ReadState.IN_PROGRESS
                         break
                     }
                 }
             } while (canParse && hasMoreData)
+
+            holder.readState = if (!hasMoreData) FileHolder.ReadState.FIN else FileHolder.ReadState.IN_PROGRESS
+            LOGGER.trace { "After read holder had status: ${holder.readState}" }
```

Comment:
I think we should move that to `processUpdates` method to keep all changes to the state close to each other